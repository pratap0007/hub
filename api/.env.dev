ENVIRONMENT="development"

POSTGRES_HOST="localhost"
POSTGRES_PORT="5432"
POSTGRES_DB="hub"
POSTGRES_USER="postgres"
POSTGRES_PASSWORD="postgres"

GH_CLIENT_ID=""
GH_CLIENT_SECRET=""
GHE_URL=""

GL_CLIENT_ID=""
GL_CLIENT_SECRET=""
GLE_URL=""

BB_CLIENT_ID=""
BB_CLIENT_SECRET=""

JWT_SIGNING_KEY=""
ACCESS_JWT_EXPIRES_IN=""
REFRESH_JWT_EXPIRES_IN=""

AUTH_BASE_URL=""

CONFIG_FILE_URL="file://../config.yaml"
