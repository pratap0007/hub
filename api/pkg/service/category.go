package hub

import (
	"context"
	"log"

	category "github.com/tektoncd/hub/api/gen/category"
)

// category service example implementation.
// The example methods log the requests and return zero values.
type categorysrvc struct {
	logger *log.Logger
}

// NewCategory returns the category service implementation.
func NewCategory(logger *log.Logger) category.Service {
	return &categorysrvc{logger}
}

// Get all Categories with their tags sorted by name
func (s *categorysrvc) All(ctx context.Context) (res []*category.Category, err error) {
	s.logger.Print("category.All")
	return
}
