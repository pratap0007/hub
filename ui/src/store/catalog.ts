import { types, Instance } from "mobx-state-tree";

export const Catalog = types
  .model({
    name: types.string,
    selected: false
  })
  .actions(self => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalog = Instance<typeof Catalog>;

export const CatalogStore = types
  .model("CatalogStore", {
    catalogList: types.array(Catalog)
  })

  .views(self => ({
    get count() {
      return self.catalogList.length;
    },
    get list() {
      const { catalogList } = self;
      return catalogList;
    },
    get selectedCatalogType() {
      const { catalogList } = self;
      return catalogList
        .filter((type: ICatalog) => type.selected)
        .map((type: ICatalog) => type.name);
    }
  }))

  .actions(self => ({
    add(item: any) {
      self.catalogList.push(item);
    },
    toggleCatalogType(kindName: string) {
      self.catalogList.forEach((kind: ICatalog) => {
        if (kind.name === kindName) {
          kind.toggle();
          return;
        }
      });
    },
    clearAll() {
      self.catalogList.map((item: ICatalog) => (item.selected = false));
    }
  }));
