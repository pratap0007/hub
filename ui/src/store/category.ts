import { AxiosError } from 'axios';
import { types, getEnv, flow, Instance } from 'mobx-state-tree';
import { Api } from '../api';

export const Category = types
  .model('Category', {
    id: types.identifierNumber,
    name: types.string,
    selected: false
  })

  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICategory = Instance<typeof Category>;
export type ICategoryStore = Instance<typeof CategoryStore>;

export const CategoryStore = types
  .model('CategoryStore', {
    items: types.map(Category),
    isLoading: true,
    err: ''
  })

  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    },

    get count() {
      return self.items.size;
    },

    get values() {
      return Array.from(self.items.values());
    },

    // This returns list of selected category's name
    get selectedByName() {
      return Array.from(self.items.values())
        .filter((c: ICategory) => c.selected)
        .reduce((acc: string[], c: ICategory) => [...acc, c.name], []);
    },

    // This returns set of selected category
    get selected(): Set<number> {
      const list: Set<number> = new Set();
      self.items.forEach((c: ICategory) => {
        if (c.selected) {
          list.add(c.id);
        }
      });
      return list;
    }
  }))

  .actions((self) => ({
    add(item: ICategory) {
      self.items.put(item);
    },

    setLoading(l: boolean) {
      self.isLoading = l;
    },

    clearSelected() {
      self.items.forEach((c) => {
        c.selected = false;
      });
    },

    toggleByName(name: string) {
      self.items.forEach((c) => {
        if (c.name === name) {
          c.selected = true;
        }
      });
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);
        const { api } = self;

        const json = yield api.categories();

        // creating the model only after the store has the tags normalized
        const categories: ICategory[] = json.data.map((c: ICategory) => ({
          id: c.id,
          name: c.name
        }));

        categories.forEach((c: ICategory) => self.add(c));
      } catch (error) {
        const err = error as AxiosError;
        self.err = err.toString();
      }

      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }));
