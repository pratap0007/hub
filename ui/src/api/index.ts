import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios.post(`${API_URL}/auth/login?code=${authCode}`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
