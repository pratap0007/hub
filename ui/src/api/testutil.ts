import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
