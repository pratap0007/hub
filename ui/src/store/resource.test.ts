import { RootStore, Resource, Catalog, kind } from './resource';
import { getSnapshot, isValidReference } from 'mobx-state-tree';
import { when } from 'mobx';
import { FakeHub } from '../api/testutil';
import fuzzysort from 'fuzzysort';
import { CategoryStore } from './category';
import { CatalogStore } from './catalog';

const TESTDATA_DIR = `${__dirname}/testdata`;
const api = new FakeHub(TESTDATA_DIR);

describe('Store Object', () => {
  it('can create a catalog object', () => {
    const store = Catalog.create({
      id: 1,
      name: 'tekton',
      type: 'community'
    });

    expect(store.name).toBe('tekton');
  });
  it('can create a kind object', () => {
    const store = kind.create({
      name: 'Task'
    });

    expect(store.name).toBe('Task');
  });
  it('can create a resource object', () => {
    const store = Resource.create({
      id: 5,
      name: 'buildah',
      catalog: '1',
      kind: 'Task',
      latestVersion: 1,
      tags: ['1'],
      rating: 5
    });

    expect(store.name).toBe('buildah');
  });
});

describe('Store functions', () => {
  it('creates a store', (done) => {
    const store = RootStore.create(
      {},
      {
        api,
        categoryStore: CategoryStore.create({}, { api }),
        catalogStore: CatalogStore.create({})
      }
    );

    expect(store.isLoading).toBe(true);

    when(
      () => !store.isLoading,
      () => {
        expect(store.isLoading).toBe(false);
        expect(store.resources.size).toBe(5);

        expect(store.catalogStore.list.get('1')?.name).toBe('tekton');

        expect(getSnapshot(store.resources)).toMatchSnapshot();

        done();
      }
    );
  });

  it('creates a catalog store', (done) => {
    const store = RootStore.create(
      {},
      {
        api,
        categoryStore: CategoryStore.create({}, { api }),
        catalogStore: CatalogStore.create({})
      }
    );
    expect(store.isLoading).toBe(true);

    when(
      () => !store.isLoading,
      () => {
        expect(store.isLoading).toBe(false);
        expect(store.resources.size).toBe(5);

        expect(getSnapshot(store.catalogStore)).toMatchSnapshot();

        done();
      }
    );
  });

  it('creates a kind store', (done) => {
    const store = RootStore.create({}, { api });

    expect(store.isLoading).toBe(true);

    when(
      () => !store.isLoading,
      () => {
        expect(store.isLoading).toBe(false);
        expect(store.resources.size).toBe(5);

        expect(getSnapshot(store.kind)).toMatchSnapshot();

        done();
      }
    );
  });

  it('can toggle a selected kind', (done) => {
    const store = RootStore.create({}, { api, catalogStore: CatalogStore.create({}) });

    expect(store.isLoading).toBe(true);

    when(
      () => !store.isLoading,
      () => {
        expect(store.isLoading).toBe(false);
        expect(store.resources.size).toBe(5);

        done();
      }
    );
  });

  fit('filter resources based on selected category and catalog', (done) => {
    const store = RootStore.create(
      {},
      {
        api,
        categoryStore: CategoryStore.create({}, { api }),
        catalogStore: CatalogStore.create({})
      }
    );
    expect(store.isLoading).toBe(true);

    when(
      () => !store.isLoading,
      () => {
        expect(store.isLoading).toBe(false);
        expect(store.resources.size).toBe(5);

        store.categoryStore.list.get('2')?.toggle();

        // store.catalogStore.list.get('1')?.toggle();
        store.setSearch('Ansible Runner');

        store.kind.get('Task')?.toggle();
        expect(store.filteredResources.length).toBe(1);
        expect(store.filteredResources[0].name).toBe('ansible-runner');

        done();
      }
    );
  });
});
