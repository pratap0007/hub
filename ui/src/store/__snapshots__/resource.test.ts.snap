// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store functions creates a kind store 1`] = `
Object {
  "items": Object {
    "Pipeline": Object {
      "name": "Pipeline",
      "selected": false,
    },
    "Task": Object {
      "name": "Task",
      "selected": false,
    },
  },
}
`;

exports[`Store functions creates a resource store 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [
      2,
    ],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      "build-tool",
    ],
    "tagsString": "build-tool",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;

exports[`Store functions fetch 0.1 version details for buildah resource 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [
      2,
    ],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      "build-tool",
    ],
    "tagsString": "build-tool",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;

exports[`Store functions makes sure to add resources with same name but from different catalog 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton-hub/Task/golang-build": Object {
    "catalog": 2,
    "categories": Array [],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton-hub/Task/golang-build",
    "tags": Array [
      1,
    ],
    "tagsString": "",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [
      2,
    ],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      "build-tool",
    ],
    "tagsString": "build-tool",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;

exports[`Store functions makes sure to not add duplicate resources 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      1,
    ],
    "tagsString": "",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;

exports[`Store functions set 0.1 as display version for buildah 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [
      2,
    ],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      "build-tool",
    ],
    "tagsString": "build-tool",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;

exports[`Store functions update versions list for buildah resource 1`] = `
Object {
  "tekton-hub/Pipeline/hub": Object {
    "catalog": 2,
    "categories": Array [
      30,
    ],
    "displayName": "Hub Pipeline",
    "displayVersion": 1,
    "id": 10,
    "kind": "Pipeline",
    "latestVersion": 1,
    "name": "hub",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton-hub/Pipeline/hub",
    "tags": Array [
      "resource",
    ],
    "tagsString": "resource",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/ansible-runner": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "Ansible Runner",
    "displayVersion": 1,
    "id": 1,
    "kind": "Task",
    "latestVersion": 1,
    "name": "ansible-runner",
    "rating": 4.5,
    "readme": "",
    "resourceKey": "tekton/Task/ansible-runner",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      1,
    ],
    "yaml": "",
  },
  "tekton/Task/aws-cli": Object {
    "catalog": 1,
    "categories": Array [
      3,
    ],
    "displayName": "aws cli",
    "displayVersion": 4,
    "id": 4,
    "kind": "Task",
    "latestVersion": 4,
    "name": "aws-cli",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/aws-cli",
    "tags": Array [
      "cli",
    ],
    "tagsString": "cli",
    "versions": Array [
      4,
    ],
    "yaml": "",
  },
  "tekton/Task/buildah": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "",
    "displayVersion": 105,
    "id": 13,
    "kind": "Task",
    "latestVersion": 105,
    "name": "buildah",
    "rating": 4,
    "readme": "",
    "resourceKey": "tekton/Task/buildah",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      105,
    ],
    "yaml": "",
  },
  "tekton/Task/golang-build": Object {
    "catalog": 1,
    "categories": Array [
      2,
    ],
    "displayName": "golang build",
    "displayVersion": 47,
    "id": 44,
    "kind": "Task",
    "latestVersion": 47,
    "name": "golang-build",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/golang-build",
    "tags": Array [
      "build-tool",
    ],
    "tagsString": "build-tool",
    "versions": Array [
      47,
    ],
    "yaml": "",
  },
  "tekton/Task/jenkins": Object {
    "catalog": 1,
    "categories": Array [
      1,
    ],
    "displayName": "jenkins operation",
    "displayVersion": 104,
    "id": 98,
    "kind": "Task",
    "latestVersion": 104,
    "name": "jenkins",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jenkins",
    "tags": Array [
      "build",
      "jenkins",
    ],
    "tagsString": "build jenkins",
    "versions": Array [
      104,
    ],
    "yaml": "",
  },
  "tekton/Task/jib-maven": Object {
    "catalog": 1,
    "categories": Array [
      19,
    ],
    "displayName": "jib maven",
    "displayVersion": 57,
    "id": 54,
    "kind": "Task",
    "latestVersion": 57,
    "name": "jib-maven",
    "rating": 5,
    "readme": "",
    "resourceKey": "tekton/Task/jib-maven",
    "tags": Array [
      "image-build",
    ],
    "tagsString": "image-build",
    "versions": Array [
      57,
    ],
    "yaml": "",
  },
}
`;
